src/model/item-data.c

```diff
@@ -249,15 +249,25 @@ void item_data_set_pos (ItemData *item_data, Coords *pos)
 void item_data_move (ItemData *item_data, const Coords *delta)
 {
 	ItemDataPriv *priv;
+	gboolean handler_connected;
 
 	g_return_if_fail (item_data != NULL);
 	g_return_if_fail (IS_ITEM_DATA (item_data));
 
 	if (delta == NULL)
 		return;
 
+	if (!delta->x && !delta->y)
+		return;
+
 	priv = item_data->priv;
 	cairo_matrix_translate (&(priv->translate), delta->x, delta->y);
+
+	handler_connected =
+	    g_signal_handler_is_connected (G_OBJECT (item_data), item_data->changed_handler_id);
```

Comment:
The check if anything is connected is unnecessary.

Reply:
That's fine, I've removed the above mentioned check.

However, there are similar checks elsewhere in the code.